testng-asserts/src/test/java/test/asserttests/AssertTest.java

```diff
@@ -563,4 +563,92 @@ public void checkArrayNotEqualsFailsWhenDifferentOrder() {
 
     assertNotEquals(array1, array2);
   }
+
+  @Test
+  public void checkMapEquals() {
+    Map<String, Object> map1 = new LinkedHashMap<>();
+    map1.put("a", "1");
+    map1.put("b", 2);
+    Map<String, Object> map2 = new LinkedHashMap<>();
+    map2.put("b", 2);
+    map2.put("a", "1");
+
+    assertMapsEqual(map1, map2);
+  }
+
+  @Test
+  public void checkMapNotEquals() {
+    Map<String, Object> map1 = new LinkedHashMap<>();
+    map1.put("a", "1");
+    map1.put("b", "2");
+    Map<String, Object> map2 = new LinkedHashMap<>();
+    map2.put("a", "1");
+    map2.put("b", 2);
+
+    assertMapsNotEqual(map1, map2);
+  }
+
+  @Test(description = "GITHUB-2913")
+  public void checkMapNotEqualsWithNull() {
+    Map<String, Object> map1 = new LinkedHashMap<>();
+    map1.put("a", 1);
+    map1.put("b", null);
+    Map<String, Object> map2 = new LinkedHashMap<>();
+    map2.put("a", 1);
+    map2.put("c", null);
+
+    assertMapsNotEqual(map1, map2);
+  }
+
+  protected void assertMapsEqual(Map m1, Map m2) {
+    assertEquals((Object) m1, (Object) m2);
+    assertEquals(m1, m2);
+
+    boolean succeeded = false;
+    try {
```

Comment:
can we refactor this to 

```
try {
     assertNotEquals((Object) m1, (Object) m2);
    Assert.fail("Maps reported as not equal when equal: " + m1 + " / " + m2);
} catch (AssertionError ignored) {}
```
This will do away with the extra variable `succeeded`

Reply:
I like the var name instead of the comment -- I've called it `expected`.

However putting the `Assert.fail` call inside the `try` will cause the `AssertionError` that _it_ throws to be ignored, won't it?, and that defeats the point of the test.  I don't know of a simple way to avoid the ugly `succeeded`, apart from introducing lambdas or splitting up the test and using `expectedExceptions` or checking the body of the failure message -- all of which I think are worse.  (But I don't feel at strongly and this is your codebase so LMK if you prefer something different!)

Reply:
(gradle insists on putting a newline between the `{}` but it saves a bit of bloat)

Reply:
You can try `{} //`

Reply:
@ahgittin - Sigh! my bad. Yes you are correct.